### PR TITLE
feat: allow changing import.meta.env

### PR DESCRIPTION
**Pull request by sheremet-va**

### CODE REVIEW

File: packages/vitest/src/node/plugins/index.ts [hunk 1]
```diff
@@ -47,6 +48,23 @@ export async function VitestPlugin(options: UserConfig = {}, ctx = new Vitest())
...
         options = deepMerge(options, viteConfig.test as any || {})
         options.api = resolveApiConfig(options)
+
+        process.env.BASE_URL = viteConfig.base
+        process.env.MODE = viteConfig.mode
+        // process.env can have only string values and will cast string on it if we pass other type,
+        // so we are making them truthy
+        process.env.PROD = viteConfig.env.PROD ? '1' : ''
+        process.env.DEV = viteConfig.env.DEV ? '1' : ''
+        process.env.SSR = '1'
```
Comment (antfu) [inline packages/vitest/src/node/plugins/index.ts:58]: We should respect the user's env if provided

Comment (sheremet-va) [reply packages/vitest/src/node/plugins/index.ts]: Added `??=`

File: packages/vitest/src/node/plugins/index.ts [hunk 2]
```diff
@@ -47,6 +48,23 @@ export async function VitestPlugin(options: UserConfig = {}, ctx = new Vitest())
         // viteConfig.test is final now, merge it for real
         options = deepMerge(options, viteConfig.test as any || {})
         options.api = resolveApiConfig(options)
+
+        process.env.BASE_URL ??= viteConfig.base
+        process.env.MODE ??= viteConfig.mode
```
Comment (antfu) [inline packages/vitest/src/node/plugins/index.ts:63]: I am not so sure about the `MODE` convention here, it feels too general to me. Do we have any references for other tools using it? or we better use `VITEST_MODE` or `VITE_MODE`

Comment (sheremet-va) [reply packages/vitest/src/node/plugins/index.ts:63]: This is what Vite uses on `import.meta.env`

Comment (sheremet-va) [reply packages/vitest/src/node/plugins/index.ts:63]: All these variables are default values on `import.meta.env` that always exist

Comment (sheremet-va) [reply packages/vitest/src/node/plugins/index.ts:63]: And we are allowing people to reassign them by making `import.meta.env` = `process.env`

we can assign them to `process.metaEnv` if you don't want to collaps with other tools, but I know this approach is wildly used now to support testing Vite in jest

**Review by antfu — APPROVED** [review]
Let's have it and see if that works for everyone

**Pull request merged**